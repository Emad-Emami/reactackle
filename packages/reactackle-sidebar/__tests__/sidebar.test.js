'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import {
    Sidebar,
    SidebarToggle,
    SidebarRegion,
} from '../src';

jest.mock('react-dom');

describe('<Sidebar/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <Sidebar />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop haveToggleButton set true', () => {
    const tree = renderer.create(
      <Sidebar haveToggleButton />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop haveToggleButton set false', () => {
    const tree = renderer.create(
      <Sidebar haveToggleButton={false} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop spread set true', () => {
    const component = renderer.create(
      <Sidebar spread />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop haveToggleButton set true' +
    ' and toggleButtonText set', () => {
    const tree = renderer.create(
      <Sidebar toggleButtonText="test" haveToggleButton />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop expanded set true', () => {
    const tree = renderer.create(
      <Sidebar expanded />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop expanded set false', () => {
    const tree = renderer.create(
      <Sidebar expanded={false} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop right set true', () => {
    const tree = renderer.create(
      <Sidebar right />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop right set false', () => {
    const tree = renderer.create(
      <Sidebar right={false} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('<SidebarToggle/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <SidebarToggle />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop toggleButtonText set', () => {
    const tree = renderer.create(
      <SidebarToggle toggleButtonText="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('one call "onToggle" event after "click"', () => {
    const mockFn = jest.fn(),
      tree = renderer.create(
        <SidebarToggle onClick={mockFn} />,
      ).toJSON();

    tree.props.onClick();

    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});

describe('<SidebarRegion/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <SidebarRegion />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop bordered', () => {
    const tree = renderer.create(
      <SidebarRegion bordered />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop spread', () => {
    const tree = renderer.create(
      <SidebarRegion spread />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop scrollable', () => {
    const tree = renderer.create(
      <SidebarRegion scrollable />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop scrollableMinHeight set 0', () => {
    const tree = renderer.create(
      <SidebarRegion scrollableMinHeight={0} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
