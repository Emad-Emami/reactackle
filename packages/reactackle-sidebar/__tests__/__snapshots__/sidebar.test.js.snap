// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> renders correctly if prop expanded set false 1`] = `
<aside
  className="SidebarStyled-huzlez iQDTlJ"
>
  <div
    className="SidebarBoxStyled-hShvpg kFprkQ"
  />
</aside>
`;

exports[`<Sidebar/> renders correctly if prop expanded set true 1`] = `
<aside
  className="SidebarStyled-huzlez gRfSoA"
>
  <div
    className="SidebarBoxStyled-hShvpg dDGNFi"
  />
</aside>
`;

exports[`<Sidebar/> renders correctly if prop haveToggleButton set false 1`] = `
<aside
  className="SidebarStyled-huzlez gRfSoA"
>
  <div
    className="SidebarBoxStyled-hShvpg dDGNFi"
  />
</aside>
`;

exports[`<Sidebar/> renders correctly if prop haveToggleButton set true 1`] = `
<aside
  className="SidebarStyled-huzlez gHnbZT"
>
  <div
    className="SidebarBoxStyled-hShvpg ctSjog"
  >
    <button
      className="SidebarToggleStyled-hXoQiO iPHaTI"
      onClick={[Function]}
    >
      <div
        className="ToggleIconStyled-kIzVex cmrxLj"
      >
        <div
          className="IconSvgStyled-cegYAl NBqjj"
          color=""
          rotate={0}
          size="normal"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <polygon
              points="13.8,18.2 7.6,12 13.8,5.8 15.2,7.2 10.4,12 15.2,16.8"
            />
          </svg>
        </div>
      </div>
      <div
        className="ToggleContentStyled-dJNHMj fMwRZj"
      >
        <div
          className="ToggleTextStyled-YrHhI jQgJxk"
        >
          Collapse
        </div>
      </div>
    </button>
  </div>
</aside>
`;

exports[`<Sidebar/> renders correctly if prop haveToggleButton set true and toggleButtonText set 1`] = `
<aside
  className="SidebarStyled-huzlez gHnbZT"
>
  <div
    className="SidebarBoxStyled-hShvpg ctSjog"
  >
    <button
      className="SidebarToggleStyled-hXoQiO iPHaTI"
      onClick={[Function]}
    >
      <div
        className="ToggleIconStyled-kIzVex cmrxLj"
      >
        <div
          className="IconSvgStyled-cegYAl NBqjj"
          color=""
          rotate={0}
          size="normal"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <polygon
              points="13.8,18.2 7.6,12 13.8,5.8 15.2,7.2 10.4,12 15.2,16.8"
            />
          </svg>
        </div>
      </div>
      <div
        className="ToggleContentStyled-dJNHMj fMwRZj"
      >
        <div
          className="ToggleTextStyled-YrHhI jQgJxk"
        >
          test
        </div>
      </div>
    </button>
  </div>
</aside>
`;

exports[`<Sidebar/> renders correctly if prop right set false 1`] = `
<aside
  className="SidebarStyled-huzlez gRfSoA"
>
  <div
    className="SidebarBoxStyled-hShvpg dDGNFi"
  />
</aside>
`;

exports[`<Sidebar/> renders correctly if prop right set true 1`] = `
<aside
  className="SidebarStyled-huzlez gRfSoA"
>
  <div
    className="SidebarBoxStyled-hShvpg dDGNFi"
  />
</aside>
`;

exports[`<Sidebar/> renders correctly if prop spread set true 1`] = `
<aside
  className="SidebarStyled-huzlez gRfSoA"
>
  <div
    className="SidebarBoxStyled-hShvpg dDGNFi"
  />
</aside>
`;

exports[`<Sidebar/> renders correctly with default props 1`] = `
<aside
  className="SidebarStyled-huzlez gRfSoA"
>
  <div
    className="SidebarBoxStyled-hShvpg dDGNFi"
  />
</aside>
`;

exports[`<SidebarRegion/> renders correctly if prop bordered 1`] = `
<div
  className="SidebarRegionStyled-bJSMUP HmlYe"
/>
`;

exports[`<SidebarRegion/> renders correctly if prop scrollable 1`] = `
<div
  className="SidebarRegionStyled-bJSMUP gJQUWD"
/>
`;

exports[`<SidebarRegion/> renders correctly if prop scrollableMinHeight set 0 1`] = `
<div
  className="SidebarRegionStyled-bJSMUP ebxVeA"
/>
`;

exports[`<SidebarRegion/> renders correctly if prop spread 1`] = `
<div
  className="SidebarRegionStyled-bJSMUP QCwEI"
/>
`;

exports[`<SidebarRegion/> renders correctly with default props 1`] = `
<div
  className="SidebarRegionStyled-bJSMUP ebxVeA"
/>
`;

exports[`<SidebarToggle/> renders correctly if prop toggleButtonText set 1`] = `
<button
  className="SidebarToggleStyled-hXoQiO iPHaTI"
  onClick={[Function]}
>
  <div
    className="ToggleIconStyled-kIzVex eBMkBa"
  >
    <div
      className="IconSvgStyled-cegYAl NBqjj"
      color=""
      rotate={0}
      size="normal"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <polygon
          points="13.8,18.2 7.6,12 13.8,5.8 15.2,7.2 10.4,12 15.2,16.8"
        />
      </svg>
    </div>
  </div>
  <div
    className="ToggleContentStyled-dJNHMj fMwRZj"
  >
    <div
      className="ToggleTextStyled-YrHhI eEqBAR"
    >
      test
    </div>
  </div>
</button>
`;

exports[`<SidebarToggle/> renders correctly with default props 1`] = `
<button
  className="SidebarToggleStyled-hXoQiO iPHaTI"
  onClick={[Function]}
>
  <div
    className="ToggleIconStyled-kIzVex eBMkBa"
  >
    <div
      className="IconSvgStyled-cegYAl NBqjj"
      color=""
      rotate={0}
      size="normal"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <polygon
          points="13.8,18.2 7.6,12 13.8,5.8 15.2,7.2 10.4,12 15.2,16.8"
        />
      </svg>
    </div>
  </div>
  <div
    className="ToggleContentStyled-dJNHMj fMwRZj"
  >
    <div
      className="ToggleTextStyled-YrHhI eEqBAR"
    >
      Collapse
    </div>
  </div>
</button>
`;
